src/logs_storage/mod.rs

```diff
@@ -1,149 +1,263 @@
-use std::sync::atomic::Ordering;
-
 use anyhow::Result;
-use atomic_enum::atomic_enum;
 use sqd_messages::{ProstMsg, QueryExecuted};
-use tokio_rusqlite::Connection;
+use tokio_rusqlite::{named_params, Connection};
 
 pub const LOGS_PER_PAGE: usize = 256;
 
 pub struct LogsStorage {
     db: Connection,
-    init_state: AtomicInitState,
 }
 
-#[atomic_enum]
-#[derive(PartialEq)]
-enum InitState {
-    NotInitialized = 0,
-    Initializing = 1,
-    Initialized = 2,
+pub enum LoadedLogs {
+    All(Vec<QueryExecuted>),
+    Partial(Vec<QueryExecuted>),
+}
+
+mod sql {
+    // timestamp is in milliseconds
+    pub const CREATE: &str = r#"
+        BEGIN;
+        CREATE TABLE IF NOT EXISTS query_logs_v2(query_id STRING PRIMARY KEY, timestamp INTEGER, log_msg BLOB);
```

Comment:
It's not possible to physically store them ordered by timestamp because incoming timestamps are not necessarily monotonic. We can add an index to make searches by `(timestamp, query_id)` fast, but we can't use timestamps themselves as row ids.